build.cake

```diff
@@ -66,8 +66,17 @@ Task("Build")
     .IsDependentOn("Restore")
     .Does(() =>
     {
-       var settings = new DotNetCoreBuildSettings { NoRestore = true, Configuration = configuration };
-       DotNetCoreBuild(solutionFullPath, settings);
+       var settings = new DotNetCoreBuildSettings
+       {
+           NoRestore = true,
+           Configuration = configuration,
+           EnvironmentVariables = new Dictionary<string, string>
+           {
+               { "Version", gitVersion.SemVer },
+               { "SourceRevisionId", gitVersion.Sha }
+           }
+        };
+        DotNetCoreBuild(solutionFullPath, settings);
```

Comment:
What's new here is the using the environment variables to pass the "Version" and "SourceRevisionId" values to MSBuild to use when generating the assembly properties.

Reply:
Very cool!